pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -824,4 +841,76 @@ public AuthenticationService getAuthenticationService() {
     public List<PersistentTopic> getAllTopicsFromNamespaceBundle(String namespace, String bundle) {
         return multiLayerTopicsMap.get(namespace).get(bundle).values();
     }
+    
+    public ZooKeeperDataCache<Map<String, String>> getDynamicConfigurationCache() {
+        return dynamicConfigurationCache;
+    }
+
+    /**
+     * Update dynamic-ServiceConfiguration with value present into zk-configuraiton-map and register listeners on
+     * dynamic-ServiceConfiguration field to take appropriate action on change of zk-configuraiton-map.
+     */
+    private void updateConfigurationAndRegisterListeners() {
+        // update ServiceConfiguration value by reading zk-configuration-map
+        updateDynamicServiceConfiguration();
+        // update brokerShutdownTimeoutMs value on listener notification
+        registerConfigurationListener("brokerShutdownTimeoutMs", new Consumer<String>() {
+            @Override
+            public void accept(String shutdownTime) {
```

Comment:
Actually, for this example, we shouldn't even need to register for notification, right? 

We should make sure that the value `ServiceConfiguration` instance is updated, and then always read it from there, instead of caching it in a variable.